### ISSUE TITLE
Turbine Presentation

### ISSUE THREAD

**Issue by mebrauns** [Microsoft/pxt-microbit#40]
the code is working in microbit.co.uk, not in codemicrobit.com

repro:

import code to microbit.co.uk

https://www.microbit.co.uk/jvkblk

import code to codemicrobit.com

https://codemicrobit.com/tgukzgtdzg

```
 let higherLimit = null;
 let voltage = null;
 let high_voltage = null;
input.onButtonPressed(Button.A, () => {
    basic.showNumber(high_voltage);
    basic.pause(2000);
});
high_voltage = 0;
higherLimit = 1024;
basic.forever(() => {
    voltage = pins.analogReadPin(AnalogPin.P0);
    if (voltage > high_voltage) {
        high_voltage = voltage;
    }
    else {
        led.plotBarGraph(voltage, higherLimit);
    }
});

```

here is a screenshot of the program working

![image](https://cloud.githubusercontent.com/assets/11206682/14364410/a28bd6ae-fcbd-11e5-8a06-01c2ef082118.png)

when this program is loaded in codemicrobit.com, the same program is NOT working

```

 let voltage = null;
 let higherLimit = null;
 let high_voltage = null;
input.onButtonPressed(Button.A, () => {
    basic.showNumber(high_voltage);
    basic.pause(2000);
});
high_voltage = 0;
higherLimit = 1024;
basic.forever(() => {
    voltage = pins.analogReadPin(AnalogPin.P0);
    if (voltage > high_voltage) {
        high_voltage = voltage;
    }
    else {
        led.plotBarGraph(voltage, higherLimit);
    }
});


```

screenshot

![image](https://cloud.githubusercontent.com/assets/11206682/14364450/dbb316cc-fcbd-11e5-8428-914d8fe60dab.png)

the error message is ...

Voltage implicitly has an error type...

![image](https://cloud.githubusercontent.com/assets/11206682/14364532/2c727dc8-fcbe-11e5-999a-c7b09747132c.png)

So I changed this to...

```
 let voltage = null;
 let higherLimit = null;
 let high_voltage = null;
input.onButtonPressed(Button.A, () => {
    basic.showNumber(high_voltage);
    basic.pause(2000);
});
high_voltage = 0;
higherLimit = 1024;
voltage = 0;
basic.forever(() => {
    voltage = pins.analogReadPin(AnalogPin.P0);
    if (voltage > high_voltage) {
        high_voltage = voltage;
    }
    else {
        led.plotBarGraph(voltage, higherLimit);
    }
});

```

possible issue: 

the variables are not updating -> variables consistently showing Null

**Comment by pelikhan** [comment]
Please add the share URL on codemicrobit.com.

**Comment by mebrauns** [comment]
Please add the share URL on codemicrobit.com.


https://codemicrobit.com/tgukzgtdzg

**Comment by mebrauns** [comment]
The code works when settting variables to 0 in JS

```
let voltage = 0;
let higherLimit = 0;
let high_voltage = 0;
input.onButtonPressed(Button.A, () => {
    basic.showNumber(high_voltage);
    basic.pause(2000);
});
high_voltage = 0;
higherLimit = 1024;
voltage = 0;
basic.forever(() => {
    voltage = pins.analogReadPin(AnalogPin.P0);
    if (voltage > high_voltage) {
        high_voltage = voltage;
    } else {
        led.plotBarGraph(voltage, higherLimit);
    }
});


```

**Comment by mebrauns** [comment]
Should I close this issue?

**Comment by pelikhan** [comment]
Is it fixed?

**Comment by mebrauns** [comment]
Still need to stream the data via serial.

**Comment by pelikhan** [comment]
Use the localhost web site or the chrome extension.

**Comment by mebrauns** [comment]
Able to stream data via serial. Good to go!

**Comment by mmoskal** [comment]
@pelikhan the problem seems to be '= null' initializers - it doesn't seem to know the variables are numbers. But maybe it's some sort of temp glitch?

**Comment by myka-x** [comment]
https://codemicrobit.com/deokbegczn

Error presents as variables being set to null, when they are numbers. 
```
 let freq = null;
 let random = null;
basic.forever(() => {
    random = Math.random(5);
    if (freq == 0) {
        random = input.rotation(Rotation.Roll);
    }
    else if (freq == 1) {
        random = input.rotation(Rotation.Pitch);
    }
    else if (freq == 2) {
        random = input.acceleration(Dimension.X);
    }
    else if (freq == 3) {
        random = input.acceleration(Dimension.Y);
    }
    else if (freq == 4) {
        random = input.acceleration(Dimension.Z);
    }
    freq = freq * 10;
    music.ringTone(freq);
    basic.pause(10);
});
```

**Comment by pelikhan** [comment]
Fix coming up.

Sent from Outlook Mobile<https://aka.ms/djubve>




On Tue, Apr 12, 2016 at 8:52 AM -0700, "myka-x" <notifications@github.com<mailto:notifications@github.com>> wrote:


https://codemicrobit.com/deokbegczn<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fcodemicrobit.com%2fdeokbegczn&data=01%7c01%7cjhalleux%40microsoft.com%7cc11c8e230e064cff574a08d362ea772b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=Gxe4Wu0IQOoi0bE6sLwyuh%2b6MiL2M5mYHyoDNQ0%2bBao%3d>

Error presents as variables being set to null, when they are numbers.

 let freq = null;
 let random = null;
basic.forever(() => {
    random = Math.random(5);
    if (freq == 0) {
        random = input.rotation(Rotation.Roll);
    }
    else if (freq == 1) {
        random = input.rotation(Rotation.Pitch);
    }
    else if (freq == 2) {
        random = input.acceleration(Dimension.X);
    }
    else if (freq == 3) {
        random = input.acceleration(Dimension.Y);
    }
    else if (freq == 4) {
        random = input.acceleration(Dimension.Z);
    }
    freq = freq * 10;
    music.ringTone(freq);
    basic.pause(10);
});


-

**Comment by pelikhan** [comment]
Fixed in PXT 0.2.60.
